lang/py/avro/errors.py

```diff
@@ -44,10 +44,18 @@ class IgnoredLogicalType(AvroWarning):
 
 class AvroTypeException(AvroException):
     """Raised when datum is not an example of schema."""
-    def __init__(self, expected_schema, datum):
-        pretty_expected = json.dumps(json.loads(str(expected_schema)), indent=2)
-        fail_msg = "The datum {} is not an example of the schema {}".format(datum, pretty_expected)
-        super(AvroTypeException, self).__init__(fail_msg)
+    def __init__(self, **kwargs):
+        if any(key not in kwargs.keys() for key in ['expected_schema', 'name', 'datum']):
```

Comment:
Oops, this is not what I meant. For now, I think what you had before was OK. We just need to recognize that these exception classes are not an external or supported API for downstream users who want to extend them.

Reply:
Agree 👍 
Reverted to its original form in https://github.com/apache/avro/pull/1156/commits/f4ad82f8afa062d5c70e154d175977b69a5ce320